MemberToMemberDiscoveryTest.formClusterWithTwoMembersTest

_master_ (commit 124c1da596a0559159d64c6afd34ed95b015ff71)

Failed on Sonar build: https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-sonar/248/testReport/com.hazelcast.spi.discovery.multicast/MemberToMemberDiscoveryTest/formClusterWithTwoMembersTest/

Stacktrace:
```
java.lang.AssertionError: Cluster size is not correct. Expected: 2, actual: 1, instance index: 0
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:1083)
	at com.hazelcast.test.HazelcastTestSupport$14.run(HazelcastTestSupport.java:1111)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1294)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1311)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1108)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1096)
	at com.hazelcast.spi.discovery.multicast.MemberToMemberDiscoveryTest.formClusterWithTwoMembersTest(MemberToMemberDiscoveryTest.java:66)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Hiccups measured while running test 'formClusterWithTwoMembersTest(com.hazelcast.spi.discovery.multicast.MemberToMemberDiscoveryTest):'
13:10:45, accumulated pauses: 1724 ms, max pause: 216 ms, pauses over 1000 ms: 0
13:10:50, accumulated pauses: 29 ms, max pause: 1 ms, pauses over 1000 ms: 0
13:10:55, accumulated pauses: 30 ms, max pause: 2 ms, pauses over 1000 ms: 0
13:11:00, accumulated pauses: 26 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:11:05, accumulated pauses: 27 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:11:10, accumulated pauses: 27 ms, max pause: 1 ms, pauses over 1000 ms: 0
13:11:15, accumulated pauses: 28 ms, max pause: 1 ms, pauses over 1000 ms: 0
13:11:20, accumulated pauses: 27 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:11:25, accumulated pauses: 26 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:11:30, accumulated pauses: 31 ms, max pause: 3 ms, pauses over 1000 ms: 0
13:11:35, accumulated pauses: 27 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:11:40, accumulated pauses: 30 ms, max pause: 1 ms, pauses over 1000 ms: 0
13:11:45, accumulated pauses: 29 ms, max pause: 2 ms, pauses over 1000 ms: 0
13:11:50, accumulated pauses: 26 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:11:55, accumulated pauses: 27 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:12:00, accumulated pauses: 27 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:12:05, accumulated pauses: 30 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:12:10, accumulated pauses: 27 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:12:15, accumulated pauses: 31 ms, max pause: 4 ms, pauses over 1000 ms: 0
13:12:20, accumulated pauses: 27 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:12:25, accumulated pauses: 26 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:12:30, accumulated pauses: 27 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:12:35, accumulated pauses: 28 ms, max pause: 1 ms, pauses over 1000 ms: 0
13:12:40, accumulated pauses: 29 ms, max pause: 1 ms, pauses over 1000 ms: 0
13:12:45, accumulated pauses: 30 ms, max pause: 1 ms, pauses over 1000 ms: 0
13:12:50, accumulated pauses: 28 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:12:55, accumulated pauses: 238 ms, max pause: 231 ms, pauses over 1000 ms: 0
```

New failure from today. There are other multicast test failures on the same day, but this one looks different. The other ones are being investigated under https://github.com/hazelcast/hazelcast-enterprise/issues/2862

http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/898/testReport/junit/com.hazelcast.spi.discovery.multicast/MemberToMemberDiscoveryTest/formClusterWithTwoMembersTest/?auto_refresh=false

**Error Message**
```
Cluster size is not correct. Expected: 2, actual: 1, instance index: 0
```
**Stacktrace**
```
java.lang.AssertionError: Cluster size is not correct. Expected: 2, actual: 1, instance index: 0
	at com.hazelcast.spi.discovery.multicast.MemberToMemberDiscoveryTest.formClusterWithTwoMembersTest(MemberToMemberDiscoveryTest.java:69)
```

The cluster should be formed with two member as the following log _captured on my machine_ shows:
```
Members {size:2, ver:2} [
	Member [10.212.134.157]:6001 - 2e654fdb-986b-4fca-b19c-1bf81293e847
	Member [10.212.134.157]:6002 - 9818fea0-a790-4597-8183-85e33a95e1de this
]
```
But the test ended up with two single-member clusters with this in the log:
```
08:36:30,954  INFO |formClusterWithTwoMembersTest| - [ClusterService] formClusterWithTwoMembersTest - [172.17.0.10]:6002 [dev] [3.12-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [172.17.0.10]:6002 - 8a906e47-5eaa-4dfe-93db-0c6771e28a3d this
]

08:36:30,954  WARN |formClusterWithTwoMembersTest| - [Node] formClusterWithTwoMembersTest - [172.17.0.10]:6002 [dev] [3.12-SNAPSHOT] Config seed port is 6001 and cluster size is 1. Some of the ports seem occupied!
```

--- a/hazelcast/src/test/java/com/hazelcast/spi/discovery/multicast/MemberToMemberDiscoveryTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/discovery/multicast/MemberToMemberDiscoveryTest.java
@@ -44,6 +44,10 @@ public class MemberToMemberDiscoveryTest extends HazelcastTestSupport {
     @Rule
     public final OverridePropertyRule overrideJoinWaitSecondsRule = set("hazelcast.wait.seconds.before.join", "10");
     @Rule
+    public final OverridePropertyRule overrideMergeFirstRunDelayRule = set("hazelcast.merge.first.run.delay.seconds", "5");
+    @Rule
+    public final OverridePropertyRule overrideMergeNextRunDelayRule = set("hazelcast.merge.next.run.delay.seconds", "5");
+    @Rule
     public final OverridePropertyRule overridePreferIpv4Rule = set("java.net.preferIPv4Stack", "true");
     @Rule
     public final OverridePropertyRule overrideHazelcastLocalAddressRule = clear("hazelcast.local.localAddress");

